src/merkle/smt/full/mod.rs

```diff
@@ -294,3 +299,62 @@ impl From<&RpoDigest> for LeafIndex<SMT_DEPTH> {
         Word::from(value).into()
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for Smt {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        let filled_leaves = self.entries().collect::<Vec<_>>();
+
+        // Write the number of filled leaves for this map
+        target.write_u8(filled_leaves.len() as u8);
```

Comment:
> but this tree uses 256 bit keys, so in theory the length can be up to 2**256 elements.

I don't understand that yet. The max length of that vector in a tree of depth 64 can be 2^64, right? The leaves always double per layer. Or is that assumption wrong?

Reply:
> note: about the usize. The way things are currently encoded in the winterfell, it will most likely work as expected. One would need to encode more than u32 to overflow the length in a u32 machine, and then the serialization just fails on that target. IMO this is not great usability, and has the potential of causing desyncs or unintentional forks, and if we want to support targets like wasm we should stay inside u32 instead. But this PR is probably not the place to start modifying that.

This might be super important for the wallet! We should create an issue

Reply:
> I don't understand that yet. The max length of that vector in a tree of depth 64 can be 2^64, right? The leaves always double per layer. Or is that assumption wrong?

The depth of the SMT is exactly `64`, but the leaves are a list of pairs `(key, value)`, and you can have up to `key` unique entries in the tree. Since `key` is a WORD and has bout 256 bits, there may be up to `2**256` entries in the SMT.